pydy/system.py

```diff
@@ -134,43 +159,6 @@ def __init__(self, eom_method, constants=None, specifieds=None,
         self._evaluate_ode_function = None
 
     @property
-    def coordinates(self):
-        """Returns a list of the symbolic functions of time representing the
-        system's generalized coordinates."""
-        if sympy_equal_to_or_newer_than('0.7.6'):
-            return self.eom_method.q[:]
-        else:
-            return self.eom_method._q
-
-    @property
-    def speeds(self):
-        """Returns a list of the symbolic functions of time representing the
-        system's generalized speeds."""
-        if sympy_equal_to_or_newer_than('0.7.6'):
-            return self.eom_method.u[:]
-        else:
-            return self.eom_method._u
-
-    @property
-    def states(self):
-        """Returns a list of the symbolic functions of time representing the
-        system's states, i.e. generalized coordinates plus the generalized
-        speeds. These are in the same order as used in integration (as
-        passed into evaluate_ode_function) and match the order of the mass
-        matrix and forcing vector.
-
-        """
-        return self.coordinates + self.speeds
-
-    @property
-    def eom_method(self):
-        """This is a sympy.physics.mechanics.KanesMethod. The method used to
-        generate the equations of motion. Read-only.
-
-        """
-        return self._eom_method
-
-    @property
```

Comment:
> So if there were a situation in which M were m x m, m < n , we could take the the first or last m states as the "dynamic" states and the remaining n-m as "kinematic." 

The nice thing about passing in the eom method is that you know exactly which of the states represent the dynamical and which represent the kinematical equations. If you simply are provided the list of states and a corresponding M and right hand side I'm not sure if there is any fool proof way to deduce which states are what. But maybe. The knowledge is already available in the *Method class, so if you output a form that doesn't have that knowledge, you have to write code that deduces this.

There are other things that System should be aware of too. Units would be nice for example and semantic meaning for each state, i.e. is it an angle, a distance, a mass, etc.

The decoupling that you are proposing is going to force use to have a complex input to System in the long run. I'm not sure why we'd want to take away the ability of System to extract any information it may want from *Method.

The rolling disc and bicycle examples show what happens when len(u) < len(q) and give more general cases of the form of these equations.